cheatsheets_excluded/Content_Security_Policy_Cheat_Sheet.md

```diff
@@ -1,203 +1,199 @@
+The most basic policy assumes:
 
-This is an example to block only passive mixed content.
+-   All resources are hosted by the same domain of the document.
+-   There are no inlines or evals for scripts and style resources.
 
-`Content-Security-Policy: script-src https:; style-src https:; object-src https:; connect-src https:; frame-src https:; `
+> `Content-Security-Policy: default-src 'self';`
 
-This is an example to block only active mixed content.
+To tighten further, one can do the following:
 
-Preventing Clickjacking
------------------------
+> `Content-Security-Policy: default-src 'none'; script-src 'self'; connect-src 'self'; img-src 'self'; style-src 'self';`
 
-The established way of preventing clickjacking involves the use of the header `X-Frame-Options` (see: [Clickjacking_Defense_Cheat_Sheet](/Clickjacking_Defense_Cheat_Sheet "wikilink")). However, CSP 2.0 has a new directive `frame-ancestors`.
+This policy allows images, scripts, AJAX, and CSS from the same origin, and does not allow any other resources to load (eg. object, frame, media, etc).
 
-To prevent all framing of your content use:
+## Mixed Content Policy
 
-`Content-Security-Policy: frame-ancestors 'none'`
+- In order to prevent mixed content (resources being loaded over http, from a document loaded over https), one can use the following directive to block mixed content.
 
-To allow for your site only, use:
+  - `Content-Security-Policy: block-all-mixed-content;`
 
-`Content-Security-Policy: frame-ancestors 'self'`
+- On the other hand, if the developer is migrating from HTTP to HTTPS, the following directive will ensure that all requests will be sent over HTTPS with no fallback to HTTP:
 
-To allow for trusted domain (my-trusty-site.com), do the following:
+  - `Content-Security-Policy: upgrade-insecure-requests;`
 
-`Content-Security-Policy: frame-ancestors my-trusty-site.com `
+If the `upgrade-insecure-requests` is set, the `block-all-mixed-content` is rendered meaningless and should be removed.
 
-A word about support. Not supported in all browsers yet, Chrome 40+ and FF 35+ support, but will also default to X-Frame-Options if it exists. Spec says, CSP should take precedence. <https://w3c.github.io/webappsec/specs/content-security-policy/#frame-ancestors-and-frame-options>
+## Preventing ClickJacking
 
-Also, keep in mind the following (from the [CSP Spec](https://w3c.github.io/webappsec/specs/content-security-policy/#frame-ancestors-and-frame-options)):
+- To prevent all framing of your content use:
+  - `Content-Security-Policy: frame-ancestors 'none';`
+- To allow for your site only, use:
+  - `Content-Security-Policy: frame-ancestors 'self';`
+- To allow for trusted domain , do the following:
+  - `Content-Security-Policy: frame-ancestors trusted.com;`
 
-`The frame-ancestors directive MUST be ignored when monitoring a policy, and when a contained in a policy defined via a meta element.`
+## Loading Files from CDN
```

Comment:
Add a sample with [SRI](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/require-sri-for) directive.

Reply:
After some rest, i think we should add it later because using the [require-sri-for](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/require-sri-for) directive will force usage of SRI even for resource (script/style) from the same domain and this, for me, has no sense because SRI has been created to protect a site against malicious modification on a linked source delivered by a CDN.

So, we must define a usage context when we will add a sample about [require-sri-for](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/require-sri-for) to not mingled the user and be realistic from a usage point of view.

What do you think?

Reply:
Yes, let's discuss for the next minor of this CS.

Reply:
Yes, we will add it in next update ofthe CS, this one is enough for the moment.